src/coreclr/vm/prestub.cpp

```diff
@@ -1059,7 +1045,444 @@ PCODE MethodDesc::JitCompileCodeLocked(PrepareCodeConfig* pConfig, JitListLockEn
+
+        return targetType;
+    }
+
+    bool TrySetTargetMethod(
+        GenerationContext& cxt,
+        LPCUTF8 methodName)
+    {
+        STANDARD_VM_CONTRACT;
+        _ASSERTE(methodName != NULL);
+        _ASSERTE(cxt.Kind == UnsafeAccessorKind::Constructor
+                || cxt.Kind == UnsafeAccessorKind::Method
+                || cxt.Kind == UnsafeAccessorKind::StaticMethod);
+
+        TypeHandle targetType = cxt.TargetType;
+        _ASSERTE(!targetType.IsTypeDesc());
+
+        CorElementType declCorType;
+        CorElementType maybeCorType;
+        TypeHandle declType;
+        TypeHandle maybeType;
+
+        // Following the iteration pattern found in MemberLoader::FindMethod().
+        // Reverse order is recommended - see comments in MemberLoader::FindMethod().
+        MethodTable::MethodIterator iter(targetType.AsMethodTable());
+        iter.MoveToEnd();
+        for (; iter.IsValid(); iter.Prev())
+        {
+            MethodDesc* curr = iter.GetDeclMethodDesc();
+
+            // Check the target and current method match static/instance state.
+            if (cxt.IsTargetStatic != (!!curr->IsStatic()))
+                continue;
+
+            // Check for matching name
+            if (strcmp(methodName, curr->GetNameThrowing()) != 0)
+                continue;
+
+            // Reset reading the declaration signature
+            cxt.DeclarationSig.Reset();
+
+            MetaSig currSig(curr);
+
+            // Validate calling convention.
+            if (cxt.DeclarationSig.GetCallingConvention() != currSig.GetCallingConvention())
+                continue;
+
+            // Validate the return type and prepare for validating
+            // the current signature's argument list.
+            UINT argCount = cxt.DeclarationSig.NumFixedArgs();
+            if (cxt.Kind == UnsafeAccessorKind::Constructor)
+            {
+                // Constructors must return void.
+                if (currSig.GetReturnType() != ELEMENT_TYPE_VOID)
+                    continue;
+            }
+            else
+            {
+                declCorType = cxt.DeclarationSig.GetReturnTypeNormalized(&declType);
+                maybeCorType = currSig.GetReturnTypeNormalized(&maybeType);
+                if (declCorType != maybeCorType)
+                    continue;
+                if (declType != maybeType)
+                    continue;
+
+                // Non-constructor accessors skip the first argument
+                // when validating the target argument list.
+                cxt.DeclarationSig.SkipArg();
+                argCount--;
+            }
+
+            // Validate argument count matches.
+            if (argCount != currSig.NumFixedArgs())
+                continue;
+
+            // Validate arguments match.
+            for (; argCount > 0; --argCount)
+            {
+                declCorType = cxt.DeclarationSig.PeekArgNormalized(&declType);
```

Comment:
If I'm reading the code right, this will only populate declType if it's a valuetype - is that correct?

Reply:
Yes. This is wrong. Thanks for calling it out.